source.js

```diff
@@ -223,7 +223,7 @@ class Source extends Resource {
 
 
       const { uri } = this
-      const stream = this.uri.includes('file://') ?
+      const stream = this.uri !== undefined && this.uri.includes('file://') ?
```

Comment:
Can you check for `null` instead? or just a truthy check

Reply:
Done!